src/pd/async/util.rs

```diff
@@ -231,18 +230,28 @@ impl<Req, Resp, F> Request<Req, Resp, F>
         self.request_sent += 1;
         debug!("request sent: {}", self.request_sent);
         let r = self.req.clone();
-        let req = (self.func)(&self.client.inner, r);
-        req.then(|resp| {
-                match resp {
-                    Ok(resp) => {
-                        self.resp = Some(Ok(resp));
-                        Ok(self)
-                    }
-                    Err(err) => {
-                        error!("request failed: {:?}", err);
-                        Err(self)
+
+        ok(self)
+            .and_then(|mut ctx| {
```

Comment:
Addressed in commit https://github.com/pingcap/tikv/pull/1974/commits/d8fb52cc037192493adc74791ac6c37e41b0796b and https://github.com/pingcap/tikv/pull/1974/commits/40227014d19563bfd1fddd77d32592c73ff67dcc